src/main/java/com/google/api/codegen/config/ApiConfig.java

```diff
@@ -199,8 +219,72 @@ public static ApiConfig createDummyApiConfig(
     return ImmutableList.copyOf(CharStreams.readLines(fileReader));
   }
 
+  private static ImmutableMap<String, CollectionConfig> createCollectionConfigs(
+      DiagCollector diagCollector, Iterable<InterfaceConfigProto> interfaceConfigProtos) {
+    LinkedHashMap<String, CollectionConfig> collectionConfigsMap = new LinkedHashMap<>();
+    for (InterfaceConfigProto interfaceConfigProto : interfaceConfigProtos) {
+      for (CollectionConfigProto collectionConfigProto :
+          interfaceConfigProto.getCollectionsList()) {
+        CollectionConfig collectionConfig =
+            CollectionConfig.createCollection(diagCollector, collectionConfigProto);
+        if (collectionConfig == null) {
+          continue;
+        }
+        if (collectionConfigsMap.containsKey(collectionConfig.getEntityName())) {
+          CollectionConfig otherConfig = collectionConfigsMap.get(collectionConfig.getEntityName());
+          if (!collectionConfig.getNamePattern().equals(otherConfig.getNamePattern())) {
+            throw new IllegalArgumentException(
+                "Inconsistent collection configs across interfaces. Entity name: "
+                    + collectionConfig.getEntityName());
```

Comment:
add a diag error and return null instead, in order to accumulate more errors for reporting to the user.